common-ui-compose/src/main/java/app/tivi/common/compose/Insets.kt

```diff
@@ -0,0 +1,300 @@
+        result = 31 * result + visible.hashCode()
+        return result
+    }
+}
+
+val InsetsAmbient = staticAmbientOf<DisplayInsets>()
+
+@Composable
+fun ProvideInsets(content: @Composable () -> Unit) {
+    val view = ViewAmbient.current
+    val density = DensityAmbient.current
+
+    val displayInsets = remember { DisplayInsets() }
+
+    onCommit(view) {
+        ViewCompat.setOnApplyWindowInsetsListener(view) { _, windowInsets ->
+            displayInsets.systemBars.updateFrom(windowInsets.systemWindowInsets, density)
+
+            // Return the unconsumed insets
+            windowInsets
+        }
+        view.doOnAttach { it.requestApplyInsets() }
+
+        onDispose {
+            ViewCompat.setOnApplyWindowInsetsListener(view, null)
+        }
+    }
+
+    Providers(InsetsAmbient provides displayInsets) {
+        content()
+    }
+}
+
+/**
+ * Selectively apply additional space which matches the width/height of any system bars present
+ * on the respective edges of the screen.
+ *
+ * @param enabled Whether to apply padding using the system bar dimensions on the respective edges.
+ * Defaults to `true`.
+ */
+fun Modifier.systemBarsPadding(enabled: Boolean = true) = composed {
+    insetsPadding(
+        insets = InsetsAmbient.current.systemBars,
+        left = enabled,
+        top = enabled,
+        right = enabled,
+        bottom = enabled
+    )
+}
+
+/**
+ * Apply additional space which matches the height of the status height along the top edge
+ * of the content.
+ */
+fun Modifier.statusBarPadding() = composed {
+    insetsPadding(insets = InsetsAmbient.current.systemBars, top = true)
+}
+
+/**
+ * Apply additional space which matches the height of the navigation bar height
+ * along the [bottom] edge of the content, and additional space which matches the width of
+ * the navigation bar on the respective [left] and [right] edges.
+ *
+ * @param bottom Whether to apply padding to the bottom edge, which matches the navigation bar
+ * height (if present) at the bottom edge of the screen. Defaults to `true`.
+ * @param left Whether to apply padding to the left edge, which matches the navigation bar width
+ * (if present) on the left edge of the screen. Defaults to `true`.
+ * @param right Whether to apply padding to the right edge, which matches the navigation bar width
+ * (if present) on the right edge of the screen. Defaults to `true`.
+ */
+fun Modifier.navigationBarPadding(
+    bottom: Boolean = true,
+    left: Boolean = true,
+    right: Boolean = true
+) = composed {
+    insetsPadding(
+        insets = InsetsAmbient.current.systemBars,
+        left = left,
+        right = right,
+        bottom = bottom
+    )
+}
+
+/**
+ * Allows conditional setting of [insets] on each dimension.
+ */
+private inline fun Modifier.insetsPadding(
+    insets: Insets,
+    left: Boolean = false,
+    top: Boolean = false,
+    right: Boolean = false,
+    bottom: Boolean = false
+) = absolutePadding(
```

Comment:
Write a private pixel-based padding modifier so that you don't have to convert back and forth from dp. Stay in pixels from the system's reported insets all the way through.

This will also let you put the insets state read in layout rather than in composition when this modifier is constructed.

Reply:
Done in the latest commit 👍